File: src/main/java/online/afeibaili/mc/MCModSearch.java
package online.afeibaili.mc;

import online.afeibaili.command.Command;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;

import java.io.IOException;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;

import static online.afeibaili.MessageHandler.COMMANDS;

public class MCModSearch {
    public static void load() {
        COMMANDS.put("搜", new Command.Builder()
                .method((param, event) -> {
                    if (param.length != 2) return "搜 [mod名|关键词]\n使用MC百科搜索";
                    try {
                        Document document = Jsoup.connect("https://search.mcmod.cn/s?key=" + URLEncoder.encode(param[1], StandardCharsets.UTF_8) + "&site=&filter=0&mold=0").get();
                        Element takeTime = document.getElementsByClass("info").get(0);
                        int startTakeChar = takeTime.toString().indexOf('[');
                        int endTakeChar = takeTime.toString().indexOf(']');

                        Elements item = document.getElementsByClass("result-item");
                        item.forEach(element -> {
                            StringBuilder message = new StringBuilder();
                            Element aTag = element.getElementsByClass("head").get(0).lastElementChild();
                            if (aTag != null) {
                                message.append("📌").append(aTag.text()).append('\n')
                                        .append("🔗").append(aTag.attr("href")).append('\n')
                                        .append("📜").append(element.getElementsByClass("body").text());
                            }
                            event.getSubject().sendMessage(message.toString());
                        });
                        try {
                            return "从MC百科查找到" + item.size() + "条结果；" + takeTime.toString().substring(startTakeChar + 1, endTakeChar);
                        } catch (StringIndexOutOfBoundsException e) {
                            return "搜不到任何信息";
                        }
                    } catch (IOException e) {
                        return "无法访问MC百科！";
                    }
                })
                .build());
    }
}

File: src/main/kotlin/online/afeibaili/module/minecraft/McModSearch.kt
package online.afeibaili.module.minecraft

import net.mamoe.mirai.message.data.ForwardMessageBuilder
import net.mamoe.mirai.message.data.PlainText
import online.afeibaili.command.Command
import online.afeibaili.command.Commands
import org.jsoup.Jsoup
import org.jsoup.nodes.Element
import java.io.IOException
import java.net.URLEncoder
import java.nio.charset.StandardCharsets
import java.util.function.Consumer

object McModSearch {
    fun load() {
        Commands.register("搜词条", Command({ param, event ->
            if (param.size != 2) return@Command "搜词条 <mod名 | 关键词>\n使用MC百科搜索"
            try {
                val document = Jsoup.connect(
                    "https://search.mcmod.cn/s?key=" + URLEncoder.encode(
                        param[1],
                        StandardCharsets.UTF_8
                    ) + "&site=&filter=0&mold=0"
                ).get()
                val takeTime = document.getElementsByClass("info").get(0)
                val startTakeChar = takeTime.toString().indexOf('[')
                val endTakeChar = takeTime.toString().indexOf(']')

                val item = document.getElementsByClass("result-item")
                val forwardMessageBuilder = ForwardMessageBuilder(event.subject)
                item.forEach(Consumer { element: Element? ->
                    val message = StringBuilder()
                    val aTag = element!!.getElementsByClass("head")[0].lastElementChild()
                    if (aTag != null) {
                        message.append("📌").append(aTag.text()).append('\n')
                            .append("🔗").append(aTag.attr("href")).append('\n')
                            .append("📜").append(element.getElementsByClass("body").text())
                    }
                    forwardMessageBuilder.add(event.subject.bot.id, "查询结果", PlainText(message.toString()))
                })
                if (forwardMessageBuilder.isEmpty()) {
                    return@Command "搜不到任何信息"
                } else event.subject.sendMessage(forwardMessageBuilder.build())

                try {
                    return@Command "从MC百科查找到" + item.size + "条结果；" + takeTime.toString()
                        .substring(startTakeChar + 1, endTakeChar)
                } catch (e: StringIndexOutOfBoundsException) {
                    return@Command "搜不到任何信息"
                }
            } catch (e: IOException) {
                return@Command "无法访问MC百科！"
            }

        }))
    }
}